sympy/integrals/integrals.py

```diff
@@ -783,7 +783,15 @@ def _eval_integral(self, f, x, meijerg=None, risch=None, manual=None,
             else:
                 if i:
                     # There was a nonelementary integral. Try integrating it.
-                    return result + i.doit(risch=False)
+
+                    # if no part of the NonElementaryIntegral is integrated by
+                    # the Risch algorithm, then use the original function to
+                    # integrate, instead of re-written one
+                    if result == 0:
+                        from sympy.integrals.risch import NonElementaryIntegral
+                        return NonElementaryIntegral(f, x).doit(risch=False)
```

Comment:
How does `NonElementaryIntegral.doit` differ from `Integral.doit`? What if `risch=False` is not added?

Reply:
> What if risch=False is not added?

`risch=True` has already been tried above. Resulting in a conclusion that the integral as a whole is `NonElementary`. Also in this `if` condition `result == 0` implies, that no part of the integral was computed by Risch algorithm.

Reply:
> How does NonElementaryIntegral.doit differ from Integral.doit?

`NonElementaryIntegral` doesn't differ in any way from using `Integral`, but I wanted to make it explicitly clear that integral has been proved to be non-elementary here. Otherwise even in working there would be no difference.

Reply:
It would keep on trying the `risch=True` first. Even below, you can see that it adds `risch=False` to since we are in a block of code, where we know that Risch algorithm, wouldn't be able 'fully' integrate the given function.

This piece of doesn't do anything fancy. Just checks whether there has been any integral computation done by Risch algorithm or not. If not then the integral with `risch=False` be tried with the original `f` (function) independent of risch from rewrites of `f`.

Reply:
Ok, thanks. I was wondering since Risch integration is the only area where the difference between elementary and nonelementary integrals is significant. So I thought that having a plain `Integral` with the original function and `risch=False` could have been enough. But maybe `NonElementaryIntegral` would be useful if the integral will remain unevaluated.